File: lib/models/organizer.dart
class Organizer {
  final String id;
  final String name;
  final String? phone;
  final String? email;

  Organizer({
    required this.id,
    required this.name,
    this.phone,
    this.email,
  });

  factory Organizer.fromJson(Map<String, dynamic> json) {
    return Organizer(
      id: json['_id'] as String,
      name: (json['name'] ?? json['orgName'] ?? '') as String,
      phone: json['phone'] as String?,
      email: json['email'] as String?,
    );
  }
}

File: event_backend/models/Organizer.js
const mongoose = require("mongoose");

const OrganizerSchema = new mongoose.Schema(
  {
    // Simple organizer entity we will reference from Event via foreign key (organizerId)
    name: { type: String, required: true, trim: true },
    phone: { type: String, trim: true },
    email: {
      type: String,
      trim: true,
      lowercase: true,
      validate: {
        validator: (v) => !v || /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(v),
        message: "Invalid email format",
      },
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Organizer", OrganizerSchema);
